python/lsst/ts/MTAOS/CollOfListOfWfErr.py

```diff
@@ -0,0 +1,182 @@
+        """
+
+        try:
+            data = self._collectionData.popleft()
+            self._collectionDataTaken.append(data)
+        except IndexError:
+            data = []
+
+        return data
+
+    def clear(self):
+        """Clear the collection."""
+
+        self._collectionData.clear()
+        self._collectionDataTaken = []
+
+    def getListOfWavefrontErrorAvgInTakenData(self):
+        """Get the list of average wavefront error in taken data.
+
+        Returns
+        -------
+        list[lsst.ts.wep.ctrlIntf.SensorWavefrontData]
+            List of average wavefront error data.
+
+        Raises
+        ------
+        RuntimeError
+            No data in the collection of taken data.
+        """
+
+        # Check there is the wavefront error data in collection or not
+        numOfElements = len(self._collectionDataTaken)
+        if (numOfElements == 0):
+            raise RuntimeError("No data in the collection of taken data.")
+
+        # Do the average of wavefont error
+        listOfWfErrAvg = self._collectionDataTaken.pop()
+
+        for wfErr in listOfWfErrAvg:
+            zk = wfErr.getAnnularZernikePoly()
+            wfErr.setAnnularZernikePoly(zk/numOfElements)
+
+        idxWfErrMiss = []
+        while self._collectionDataTaken:
+
+            listOfWfErrNext = self._collectionDataTaken.pop()
+            for idx, wfErr in enumerate(listOfWfErrAvg):
+                sensorId = wfErr.getSensorId()
+                wfErrNext = self._getSensorWavefrontDataInList(listOfWfErrNext,
+                                                               sensorId)
+
+                if (wfErrNext is not None):
+                    zk = wfErr.getAnnularZernikePoly()
+                    zkNext = wfErrNext.getAnnularZernikePoly()
+                    wfErr.setAnnularZernikePoly(zk + zkNext/numOfElements)
+                else:
+                    # Collect the index of wavefront error data if there is the
+                    # data missing
+                    idxWfErrMiss.append(idx)
+
+        # Get the unique list of index of wavefront error data that has the
+        # missing data.
+        idxWfErrMissUnique = np.unique(np.array(idxWfErrMiss)).tolist()
+
+        # Only return the value that is not in the above list
+        return [listOfWfErrAvg[idx] for idx in range(len(listOfWfErrAvg))
+                if idx not in idxWfErrMissUnique]
+
+    def _getSensorWavefrontDataInList(self, listOfWfErr, sensorId):
+        """Get the sensor wavefront data in the list.
+
+        Parameters
+        ----------
+        listOfWfErr : list[lsst.ts.wep.ctrlIntf.SensorWavefrontData]
+            List of wavefront error data.
+        sensorId : int
+            Sensor Id.
+
+        Returns
+        -------
+        lsst.ts.wep.ctrlIntf.SensorWavefrontData
+            Sensor wavefront data. Return None if not find.
+        """
+
+        for wfErr in listOfWfErr:
+            if (wfErr.getSensorId() == sensorId):
```

Comment:
You might want to drop (), but that's a minor comment. Python doesn't need them. Just 
```python
if wfErr.getSensorId() == sensorId:
  ..code..
```

Reply:
Fixed in commit: 73299e9. Thanks!